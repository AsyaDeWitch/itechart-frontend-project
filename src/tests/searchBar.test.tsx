import { unmountComponentAtNode } from "react-dom";
import renderer from "react-test-renderer";
import SearchBar from "../components/home/elements/searchBar/searchBar";

let container: Element;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container?.remove();
});

describe("Search bar", () => {
  test("Snapshot test should be the same with previous version", () => {
    const mockOnChangeFunction = jest.fn();

    const component = renderer.create(<SearchBar onChange={mockOnChangeFunction} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
