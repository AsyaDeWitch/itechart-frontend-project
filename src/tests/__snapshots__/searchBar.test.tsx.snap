// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar Snapshot test should be the same with previous version 1`] = `
<input
  className="searchBar"
  id="home-search"
  onChange={[MockFunction]}
  placeholder="Search"
  type="text"
/>
`;
